src/kv.c

```diff
@@ -1200,7 +1218,14 @@ kvStore_Keys(kvKeysList *list, kvStore *kv, kvWatchOptions *opts)
     if (o.Timeout > 0)
         timeout = o.Timeout;
 
-    s = kvStore_WatchAll(&w, kv, &o);
+    if (filters == NULL) {
+        s = kvStore_WatchAll(&w, kv, &o);
+    } else {
+        // Deliver policy gets set to DeliverLastPerSubject
+        o.IncludeHistory = false;
```

Comment:
That I don't understand. `kvStore_WatchAll()` is already calling `kvStore_WatchMulti()` with the ">" filter, and there was no need to add anything to it.